iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -279,23 +451,24 @@ void PortManager::handleInterfaces() noexcept
                 }
             }
         }
-        // also forward services from service registry
-        /// @todo #415 do we still need this? yes but return a copy here to be stored in shared memory via new
-        /// StatusPort's
-        /// @todo iox-#27 I guess this was necessary since a service could be offered via ServiceDiscovery;
-        /// this was removed and I somehow have the feeling this breaks the interface ports with the changes from this
-        /// PR if the CaproServiceType is something different than NON
-        auto serviceVector = m_serviceRegistry.getServices();
-
-        caproMessage.m_serviceType = capro::CaproServiceType::NONE;
-
-        for (auto const& element : serviceVector)
+        // provide offer information from all active server ports to all new interfaces
+        caproMessage.m_serviceType = capro::CaproServiceType::SERVER;
+        for (auto serverPortData : m_portPool->getServerPortDataList())
         {
-            caproMessage.m_serviceDescription = element.serviceDescription;
-
-            for (auto& interfacePortData : interfacePortsForInitialForwarding)
+            popo::ServerPortUser serverPort(*serverPortData);
+            if (serverPort.isOffered())
             {
-                popo::InterfacePort(interfacePortData).dispatchCaProMessage(caproMessage);
+                caproMessage.m_serviceDescription = serverPort.getCaProServiceDescription();
+                for (auto& interfacePortData : interfacePortsForInitialForwarding)
+                {
+                    auto interfacePort = popo::InterfacePort(interfacePortData);
+                    // do not offer on same interface
+                    if (serverPort.getCaProServiceDescription().getSourceInterface()
+                        != interfacePort.getCaProServiceDescription().getSourceInterface())
+                    {
+                        interfacePort.dispatchCaProMessage(caproMessage);
+                    }
+                }
```

Comment:
@mossmaurice to continue our discussion from the previous PR. I think I can remove the service discovery items since all the publisher and server are already forwarded to the new interface ports